src/server/controllers/collection.controller.ts

```diff
@@ -0,0 +1,53 @@
+import { Context } from '~/server/createContext';
+import { throwDbError } from '~/server/utils/errorHandling';
+import {
+  AddCollectionItemInput,
+  GetAllUserCollectionsInputSchema,
+} from '~/server/schema/collection.schema';
+import {
+  addCollectionItems,
+  getUserCollectionsWithPermissions,
+} from '~/server/services/collection.service';
+
+export const getAllUserCollectionsHandler = async ({
+  ctx,
+  input,
+}: {
+  ctx: DeepNonNullable<Context>;
+  input: GetAllUserCollectionsInputSchema;
+}) => {
+  const { user } = ctx;
+  const { permission } = input;
+
+  try {
+    const collections = await getUserCollectionsWithPermissions({
+      user,
+      permissions: [permission],
+      select: {
+        id: true,
+        name: true,
+        description: true,
+        coverImage: true,
+      },
+    });
+
+    return collections;
+  } catch (error) {
+    throwDbError(error);
```

Comment:
We have to actually throw in here to get the right return type for this controller
```suggestion
    throw throwDbError(error);
```